The job **`x86_64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8559441635/job/23456200456) [(plain)](https://github.com/rust-lang/rust/commit/c6d9e7d8abad316bf77c864a5ece6e7f339ff4c5/checks/23456200456/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[command]C:\Windows\system32\cmd.exe /D /S /C C:\a\_temp\setup-msys2\msys2.cmd -c "'paccache' '-r' '-f' '-u' '-k0'"
==> no candidate packages found for pruning
[command]C:\Windows\system32\cmd.exe /D /S /C C:\a\_temp\setup-msys2\msys2.cmd -c "'paccache' '-r' '-f' '-k1'"
==> no candidate packages found for pruning
[command]"C:\Program Files\Git\usr\bin\tar.exe" --posix -cf cache.tzst --exclude cache.tzst -P -C C:/a/rust/rust --files-from manifest.txt --force-local --use-compress-program "zstd -T0"
Cache saved successfully
Cache saved as ID 177 using key msys2-pkgs-upd:false-conf:1d008dec-files:c7bd2894f33f12029617f9e89abb697412345fee
[command]C:\Windows\system32\cmd.exe /D /S /C C:\a\_temp\setup-msys2\msys2.cmd -c "'pacman' '--noconfirm' '-Scc'"

---
  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 72ff5730 Merge c6d9e7d8abad316bf77c864a5ece6e7f339ff4c5 into 0fd571286ef6df8a6cce06d432013239a8c0665f
[command]"C:\Program Files\Git\bin\git.exe" log -1 --format='%H'
'72ff57301262405270686dca0b6ad2784add9472'
'72ff57301262405270686dca0b6ad2784add9472'
##[group]Run echo "[CI_PR_NUMBER=$num]"
echo "[CI_PR_NUMBER=$num]"
---
error: the compiler unexpectedly panicked. this is a bug.

note: using internal features is not supported and expected to cause internal compiler errors when used incorrectly

note: please attach the file at `C:\a\rust\rust\compiler\rustc_codegen_cranelift\rustc-ice-2024-04-04T19_13_50-5812.txt` to your bug report

note: compiler flags: -C panic=abort -Z panic-abort-tests -Z codegen-backend=cranelift -C symbol-mangling-version=v0 -Z unstable-options -Z macro-backtrace -C split-debuginfo=packed -C target-feature=+crt-static -C debuginfo=2 -C panic=abort -Z unstable-options --crate-type lib,dylib -Z force-unstable-if-unmarked
query stack during panic:
end of query stack
end of query stack
"C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\cg_clif\\dist\\rustc-clif.exe" "-Csymbol-mangling-version=v0" "-Zunstable-options" "--check-cfg=cfg(bootstrap)" "--check-cfg=cfg(parallel_compiler)" "-Zmacro-backtrace" "-Csplit-debuginfo=packed" "-Ctarget-feature=+crt-static" "--cfg=parallel_compiler" "-L" "crate=C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\cg_clif\\build\\example" "--out-dir" "C:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage1-tools\\cg_clif\\build\\example" "-Cdebuginfo=2" "--target" "x86_64-pc-windows-msvc" "-Cpanic=abort" "-Zunstable-options" "--check-cfg=cfg(no_unstable_features)" "--check-cfg=cfg(jit)" "example/mini_core.rs" "--crate-type" "lib,dylib" "--cfg" "no_unstable_features" exited with status ExitStatus(ExitStatus(2147483651))


Command did not execute successfully.
Expected success, got: exit code: 1
Expected success, got: exit code: 1
Add `-v` to see more details.

Build completed unsuccessfully in 0:41:09
make: *** [Makefile:73: ci-msvc-ps1] Error 1
  network time: Thu, 04 Apr 2024 19:13:51 GMT
##[error]Process completed with exit code 2.
Post job cleanup.
[command]"C:\Program Files\Git\bin\git.exe" version
```

</details>

--- a/compiler/rustc_codegen_ssa/src/back/archive.rs
+++ b/compiler/rustc_codegen_ssa/src/back/archive.rs
@@ -231,7 +231,11 @@ impl<'a> ArArchiveBuilder<'a> {
             "gnu" => ArchiveKind::Gnu,
             "bsd" => ArchiveKind::Bsd,
             "darwin" => ArchiveKind::Darwin,
-            "coff" => ArchiveKind::Coff,
+            "coff" => {
+                // FIXME: ar_archive_writer doesn't support COFF archives yet.
+                // https://github.com/rust-lang/ar_archive_writer/issues/9
+                ArchiveKind::Gnu
+            }
             "aix_big" => ArchiveKind::AixBig,
             kind => {
                 self.sess.dcx().emit_fatal(UnknownArchiveKind { kind });

--- a/compiler/rustc_target/src/spec/base/msvc.rs
+++ b/compiler/rustc_target/src/spec/base/msvc.rs
@@ -14,6 +14,7 @@ pub fn opts() -> TargetOptions {
         pre_link_args,
         abi_return_struct_as_int: true,
         emit_debug_gdb_scripts: false,
+        archive_format: "coff".into(),
 
         // Currently this is the only supported method of debuginfo on MSVC
         // where `*.pdb` files show up next to the final artifact.

